driver/src/orderbook/streamed/state.rs

```diff
@@ -83,17 +123,20 @@ impl State {
     /// `block_batch_id` is the current batch based on the timestamp of the block that contains the
     ///  event.
     pub fn apply_event(&mut self, event: &Event, block_batch_id: BatchId) -> Result<(), Error> {
+        self.apply_pending_solution_if_needed(block_batch_id);
```

Comment:
What happens if there is a batch without an event at all? Would we then not be able to get the account state as nothing would trigger the apply_pending_solution if needed?

Generally, it would be great if we could query the currently foreseen balance state for a future batch so that we can optimistically fetch state before the batch is finalized.